elastic-buffer-sim/src/Bittide/Simulate.hs

```diff
@@ -85,21 +85,31 @@ tunableClockGen settlePeriod periodOffset stepSize _reset speedChange =
     in
       newPeriod :- go newSettleCounter newPeriod scs
 
+-- | This wrapper disables reads or writes when the elastic buffer
+-- over-/underflows, as appropriate.
+--
+-- It also censors 'DataCount' after an over-/underflow, so that we do not take
+-- measurements from an elastic buffer until it has settled back to its
+-- midpoint.
 ebController ::
   forall readDom writeDom.
   (KnownDomain readDom, KnownDomain writeDom) =>
   ElasticBufferSize ->
   Clock readDom ->
   Clock writeDom ->
   Signal readDom (Maybe DataCount)
-ebController size clkRead clkWrite = go <$> rdEna <*> outRd
+ebController size clkRead clkWrite =
+  go <$> rdToggle <*> outRd <*> overflowRd
  where
-  (outRd, outWr) = elasticBuffer size clkRead clkWrite rdEna wrEna
+  (outRd, outWr) = elasticBuffer size clkRead clkWrite rdToggle wrToggle
+
+  go True (dc, False) False = Just dc
+  go _ _ _ = Nothing
 
-  go True (dc, False) = Just dc
-  go _ _ = Nothing
+  overflowRd =
+    dualFlipFlopSynchronizer clkWrite clkRead resetGen enableGen False (snd <$> outWr)
```

Comment:
This is something I hadn't considered yet. You probably want to drain the FIFO completely and then start reading again at its midpoint, because it's impossible to predict the latency write-to-read...

Reply:
Sure.